transactions/user/transfer_moment_v3_sale.cdc

```diff
@@ -0,0 +1,43 @@
+import NonFungibleToken from 0xNFTADDRESS
+import TopShot from 0xTOPSHOTADDRESS
+import TopShotMarketV3 from 0xMARKETV3ADDRESS
+
+// This transaction transfers a moment to a recipient
+// and cancels the sale in the V3 collection if it exists
+
+// Parameters:
+//
+// recipient: The Flow address of the account to receive the moment.
+// withdrawID: The id of the moment to be transferred
+
+transaction(recipient: Address, withdrawID: UInt64) {
+
+    // local variable for storing the transferred token
+    let transferToken: @NonFungibleToken.NFT
+    
+    prepare(acct: AuthAccount) {
+
+        // borrow a reference to the owner's collection
+        let collectionRef = acct.borrow<&TopShot.Collection>(from: /storage/MomentCollection)
+            ?? panic("Could not borrow a reference to the stored Moment collection")
+        
+        // withdraw the NFT
+        self.transferToken <- collectionRef.withdraw(withdrawID: withdrawID)
+
+        if let saleRef = acct.borrow<&TopShotMarketV3.SaleCollection>(from: TopShotMarketV3.marketStoragePath) {
+            saleRef.cancelSale(tokenID: withdrawID)
+        }
```

Comment:
Are we already cancelling the sale before a transaction here?

Reply:
Yeah, this is transaction that cancels the sale at the same time as the transfer